A/AWSCLI/build_tarballs.jl

```diff
@@ -0,0 +1,40 @@
+using BinaryBuilder
+
+name = "AWSCLI"
+version = v"2.17.35"
+
+sources = [
+    GitSource("https://github.com/aws/aws-cli.git",
+              "a55557d53742d03823bba42c42b3259a086ccf7f"),
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/aws-cli
+
+export CMAKE_INSTALL_PREFIX="${prefix}"
+export CMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}"
+
+# Breaking change in Clang 16 causes failure while building dependencies on macOS and FreeBSD
+# See https://discourse.llvm.org/t/clang-16-notice-of-potentially-breaking-changes/65562
+export CFLAGS="${CFLAGS} -Wno-error=incompatible-function-pointer-types"
+
+PYTHON="${bindir}/python3" ./configure --prefix=${prefix} --with-download-deps --with-install-type=portable-exe
```

Comment:
If they need to run code with `PYTHON`, this isn't going to work because `${bindir}/python3` is target's python and (in general) we can't run it.  There should be a working `python3` in the system, but I don't know what they want to do with python at all.

Reply:
I had been using the system Python but ran into an issue I didn't understand. Hadn't occurred to me that it'd need to run Python, that definitely won't work. 🤦

Reply:
On the few platforms where we can actually run target code (e.g. https://buildkite.com/julialang/yggdrasil/builds/12639#01917bcf-56fa-4c1b-b4ab-d0d3f8f00321)
```
[20:39:03] configure: error: AWS CLI requires a Python interpreter compiled with sqlite3. /workspace/destdir/bin/python3 does not have sqlite3.
```

This package looks like a Python package, just with a fancy build system.  I don't really think we aim to be a Python package manager.

Reply:
The developers consider the fact that it's written in Python to be an "implementation detail," which is why they don't distribute it via PyPI; it's intended to just be a command line tool that happens to currently be written in Python rather than an actual Python package.

Reply:
That said, that implementation detail probably means we can't build it with BinaryBuilder. Oh well.